src/python/pants/engine/internals/scheduler.py

```diff
@@ -176,29 +205,25 @@ def __init__(
             self.visualize_rule_graph_to_file(os.path.join(self._visualize_to_dir, rule_graph_name))
 
         if validate_reachability:
-            self._native.lib.validate_reachability(self._scheduler)
+            native_engine.validate_reachability(self.py_scheduler)
 
-    def graph_trace(self, session, execution_request):
-        with temporary_file_path() as path:
-            self._native.lib.graph_trace(self._scheduler, session, execution_request, path)
-            with open(path, "r") as fd:
-                for line in fd.readlines():
-                    yield line.rstrip()
+    @property
+    def py_scheduler(self) -> PyScheduler:
+        return self._py_scheduler
 
-    def _to_params_list(self, subject_or_params):
+    def _to_params_list(self, subject_or_params: Any | Params) -> Sequence[Any]:
         if isinstance(subject_or_params, Params):
             return subject_or_params.params
         return [subject_or_params]
 
-    def visualize_graph_to_file(self, session, filename):
-        self._native.lib.graph_visualize(self._scheduler, session, filename)
```

Comment:
I removed any `Scheduler` method that takes `session` as a param and put it directly on `SchedulerSession`. This removes some additional boilerplate, particularly with type hints.